src/odemis/dataio/test/hdf5_test.py

```diff
@@ -426,13 +429,14 @@ def testMetadata(self):
         # Try reading the metadata using the hdf5 module
         rdata = hdf5.read_data(FILENAME)
         self.assertEqual(len(rdata), 1)
-
         for im in rdata:
             self.assertEqual(im.metadata[model.MD_DESCRIPTION], metadata[model.MD_DESCRIPTION])
             self.assertEqual(im.metadata[model.MD_POS], metadata[model.MD_POS])
             self.assertEqual(im.metadata[model.MD_PIXEL_SIZE], metadata[model.MD_PIXEL_SIZE])
             self.assertEqual(im.metadata[model.MD_ACQ_DATE], metadata[model.MD_ACQ_DATE])
             self.assertEqual(im.metadata[model.MD_EXP_TIME], metadata[model.MD_EXP_TIME])
+            self.assertEqual(sorted(im.metadata[model.MD_EXTRA_SETTINGS].items()),
+                             sorted(metadata[model.MD_EXTRA_SETTINGS].items()))
```

Comment:
I don't understand why you need so use `sorted(...items())`. It's a dict, so there is no order anyway. Or is there a more hidden reason?

Reply:
I didn't realize `assertEqual` could also compare dictionaries (because they are not ordered), but actually it works just fine. I'll change it.